lib/host/wasi/wasifunc.cpp

```diff
@@ -1847,5 +1847,145 @@ Expect<uint32_t> WasiSockShutdown::body(Runtime::Instance::MemoryInstance *,
   return __WASI_ERRNO_SUCCESS;
 }
 
+Expect<uint32_t>
+WasiGetAddrinfo::body(Runtime::Instance::MemoryInstance *MemInst,
+                      uint32_t NodePtr, uint32_t NodeLen, uint32_t ServicePtr,
+                      uint32_t ServiceLen, uint32_t HintsPtr, uint32_t ResPtr,
+                      uint32_t MaxResLength, uint32_t ResLengthPtr) {
+  /// Check memory instance from module.
+  if (MemInst == nullptr) {
+    return __WASI_ERRNO_FAULT;
+  }
+  const char *Node = nullptr;
+  if (NodeLen != 0) {
+    Node = MemInst->getPointer<const char *>(NodePtr, NodeLen);
+  }
+  const char *Service = nullptr;
+  if (ServiceLen != 0) {
+    Service = MemInst->getPointer<const char *>(ServicePtr, ServiceLen);
+  }
+  // service and node can not be nullptr at the same time
+  if (Service == nullptr && Node == nullptr) {
+    return __WASI_ERRNO_FAULT;
+  }
+  if (unlikely(MaxResLength > WASI::addrinfoArrayMax)) {
+    return __WASI_ERRNO_INVAL;
+  }
+  auto *const Hint = MemInst->getPointer<const struct __wasi_addrinfo_t *>(
+      HintsPtr, sizeof(struct __wasi_addrinfo_t));
+  auto *const ResLength = MemInst->getPointer<__wasi_size_t *>(
+      ResLengthPtr, sizeof(__wasi_size_t *));
+
+  struct addrinfo *TmpAddrinfoRes = nullptr;
```

Comment:
Please use `Sys` to replace `Tmp`. Which means these information is from system call.